### PR TITLE
[BPK-1289] Add flow to bpk-component-input

### PR DESCRIPTION
**Pull request by k0nserv**

### CODE REVIEW

**Comment by matthewdavidson** [comment]
I have an idea to avoid a breaking change, add this to the `index.js` entry point file:
```js
// remove this on the next major
export const INPUT_TYPES = { TEXT: 'text', blah, blah, blah, ...TYPES };
```

This allows us to use the flow goodness inside the component, whilst maintaining the API for consumers.

We can remove the hackyness on the next warranted major

**Comment by matthewdavidson** [comment]
Can you update all usages to use the new lowercase syntax? `readme.md`'s and usages in `bpk-docs` etc.

File: unreleased.md [hunk 1]
```diff
@@ -3,3 +3,7 @@
 **Added:**
 - react-native-bpk-component-pagination-dots:
   - Introducing the React Native pagination dots component.
+
+**Changed:**
```
Comment (matthewdavidson) [inline unreleased.md]: I don't think this warrants a changelog entry now. Nothing has changed from a consumer POV.

Comment (k0nserv) [reply unreleased.md]: We've deprecated the old keys and intend to remove them. That's definitely something we should communicate IMO

**Pull request merged**